lib/Sema/TypeCheckStmt.cpp

```diff
@@ -82,7 +82,13 @@ namespace {
         if (CE->getParent() == ParentDC
             && CE->getDiscriminator() != AutoClosureExpr::InvalidDiscriminator)
           return { false, E };
-        
+        // Stop the recursion if the autoclosure is already checked in
+        // default initializer.
+        if (auto *initContext = dyn_cast<PatternBindingInitializer>(ParentDC))
+          if (initContext->getBinding()->isInitializerChecked(0) &&
```

Comment:
It is incorrect to use `0` here, instead `initContext->getBindingIndex()` should be used to find correct location.

Reply:
Many thanks for the quick review and explanation!
I will push the commit after local testing is passed!

Reply:
No worries!